go-controller/pkg/ovn/defaultcontroller.go

```diff
@@ -0,0 +1,845 @@
+package ovn
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sync"
+
+	ocpcloudnetworkapi "github.com/openshift/api/cloudnetwork/v1"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	egressfirewall "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressfirewall/v1"
+	egressipv1 "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressip/v1"
+	egressqoslisters "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressqos/v1/apis/listers/egressqos/v1"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/metrics"
+	addressset "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/address_set"
+	egresssvc "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/controller/egress_services"
+	svccontroller "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/controller/services"
+	lsm "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/logical_switch_manager"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/subnetallocator"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/retry"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/syncmap"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	kapi "k8s.io/api/core/v1"
+	knet "k8s.io/api/networking/v1"
+	ktypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/informers"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+type defaultNetworkControllerEventHandler struct {
+	masterEventHandler masterEventHandler
```

Comment:
minor nit: could embed this so then you can just do later:
```
func (h *defaultNetworkControllerEventHandler) AreResourcesEqual(obj1, obj2 interface{}) (bool, error) {
	return h.areResourcesEqual(h.objType, obj1, obj2)
}
```

Reply:
it was embeded, but changed it based on @jcaamano 's comments: https://github.com/ovn-org/ovn-kubernetes/pull/3258#discussion_r1023925049

Reply:
Yes, I suggested not to embed so that the utility methods cannot be as easily seen from the outside, given that their intention is to be used internally and their similarity with their exported counterparts.
If this were sitting on its own package, this wouldn't be as relevant since the methods are not exported. But since ovn package is a big box of things, I thought this would help with clarity.